### PR TITLE
Update to Laravel 10

### PR DESCRIPTION
**Pull request by jxjj**
Not the latest version, but close. There's an open issue with 10.43.0 which current breaks `after:date` validation.

Having some weird issue deploying to dev: deployer can't seem to find the remote branch on the host. I'll check it out more tomorrow.

### CODE REVIEW

File: app/Group.php [hunk 1]
```diff
@@ -19,7 +19,9 @@ class Group extends Model implements AuditableContract {
 
     protected $casts = [
         'include_child_groups' => 'boolean',
-        'show_unit' => 'boolean'
+        'show_unit' => 'boolean',
+        'start_date' => 'datetime',
+        'end_date' => 'datetime'
```
Comment (cmcfadden) [inline app/Group.php:24]: does deleted_at not need this cast too?

File: app/Leave.php [hunk 2]
```diff
@@ -22,11 +22,9 @@ class Leave extends Model implements Auditable {
...
     ];
 
-    // The attributes that should be mutated to dates.
-    protected $dates = [
-        'start_date',
-        'end_date',
-        'deleted_at',
+    public $casts = [
+        'start_date' => 'datetime',
+        'end_date' => 'datetime',
```
Comment (cmcfadden) [inline app/Leave.php:27]: same question here - is the cast handled automatically by the soft deletes?

**Review by cmcfadden — APPROVED** [review]
looks good

File: app/Group.php (hunk 1 above)
Comment (jxjj) [reply app/Group.php:24]: I don't think so. [`runSoftDelete()`](https://github.com/laravel/framework/blob/9448e01699d43761e8bcf3d2ff53d0e3efb37432/src/Illuminate/Database/Eloquent/SoftDeletes.php#L98) invokes `freshTimestamp()` like the other timestamp columns (which in turn uses Carbon).

File: app/Leave.php (hunk 2 above)
Comment (jxjj) [reply app/Leave.php:27]: Same.

**Comment by jxjj** [comment]
Deployed to dev now.

**Pull request merged**